components/director/internal/domain/formationassignment/notifications.go

```diff
@@ -344,3 +358,29 @@ func convertFormationAssignmentFromModel(formationAssignment *model.FormationAss
 		Value:       config,
 	}
 }
+
+func (fan *formationAssignmentNotificationService) extractCustomerTenantContext(ctx context.Context, internalTenantID string) (*webhook.CustomerTenantContext, error) {
+	tenantObject, err := fan.tenantRepository.Get(ctx, internalTenantID)
+	if err != nil {
+		return nil, err
+	}
+
+	customerID, err := fan.tenantRepository.GetCustomerIDParentRecursively(ctx, internalTenantID)
```

Comment:
The customerID may be empty string. Should we check for that and return error if empty?

Reply:
No, this is the desired state. If `customerID` is empty, that means that the parent of `internalTenantID ` is not of type customer. In the notification payload we want to populate only the customer parent, so if there isn't such, we leave it empty.